runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala

```diff
@@ -51,22 +52,33 @@ object AkkaHttpClientUtils {
     // Configuring compression per call could be a future power API feature.
     implicit val writer = GrpcProtocolNative.newWriter(Identity)
 
-    // TODO FIXME discovery, loadbalancing etc
-    val host = settings.serviceName
+    val target = {
+      // TODO FIXME discovery, loadbalancing etc
+      // https://github.com/akka/akka-grpc/issues/1196
+      // https://github.com/akka/akka-grpc/issues/1197
+      val resolved = settings.serviceDiscovery.lookup(settings.serviceName, 10.seconds)
+      Await.result(resolved, 10.seconds).addresses.head
+    }
 
     val clientConnectionSettings = settings.overrideAuthority match {
       case None => ClientConnectionSettings(sys)
       case Some(authority) =>
         ClientConnectionSettings(sys).withTransport(ClientTransport.withCustomResolver((host, port) => {
           assert(host == authority)
           assert(port == settings.defaultPort)
-          Future.successful(new InetSocketAddress(settings.serviceName, settings.defaultPort))
+          target.address match {
+            case Some(address) =>
+              Future.successful(new InetSocketAddress(address, target.port.getOrElse(settings.defaultPort)))
+            case None =>
+              Future.successful(new InetSocketAddress(target.host, target.port.getOrElse(settings.defaultPort)))
+          }
+
         }))
     }
 
     val builder = Http()
-      .connectionTo(settings.overrideAuthority.getOrElse(host))
-      .toPort(settings.defaultPort)
+      .connectionTo(settings.overrideAuthority.getOrElse(target.host))
+      .toPort(target.port.getOrElse(settings.defaultPort))
       .withClientConnectionSettings(clientConnectionSettings)
```

Comment:
If `settings.overrideAuthority` and `target.port` are defined, this builder would use `Http().connectionTo(settings.overrideAuthority).toPort(target.port)`.

Couldn't that trip the assertion on line 68? (`assert(port == settings.defaultPort)`)

I don't think there's a way to trip the other assertion with the current code.

This class is growing a bit. Maybe we could use this PR to extract the creation of the `builder` into a dedicated class/object that's easier to test/inject/extend?

Reply:
> Couldn't that trip the assertion on line 68?

Good eye, that needed an update as well indeed

> This class is growing a bit. Maybe we could use this PR to extract the creation of the `builder` into a dedicated class/object that's easier to test/inject/extend?

I agree, we should probably move the `InternalChannel` anonymous subclass into its own file eventually, but since the connection management / loadbalancing might still cause code to move between the channel and the initialization logic I think we should do it when/after we implement that.